File: android/src/main/java/com/cactuslab/capacitor/nativenavigation/types/ModalPresentationStyle.java
package com.cactuslab.capacitor.nativenavigation.types;

import com.google.gson.annotations.SerializedName;

public enum ModalPresentationStyle {
    @SerializedName("fullScreen")
    FULLSCREEN,
    @SerializedName("pageSheet")
    PAGE_SHEET,
    @SerializedName("formSheet")
    FORM_SHEET
}

File: android/src/main/java/com/cactuslab/capacitor/nativenavigation/types/ModalPresentationStyle.kt
package com.cactuslab.capacitor.nativenavigation.types

enum class ModalPresentationStyle(val id: String) {
    FULLSCREEN("fullscreen"), PAGE_SHEET("pageSheet"), FORM_SHEET("formSheet");

    companion object {
        operator fun get(id: String?): ModalPresentationStyle? {
            for (value in values()) {
                if (value.id.contentEquals(id)) {
                    return value
                }
            }
            return null
        }
    }
}